packages/dds/tree/src/schema-stored/Stored and View Schema.md

```diff
@@ -35,104 +34,116 @@ Places we can store stored schema information:
+
+    They may additionally use schema-on-read to enforce/fix some invariants the stored schema system isn't expressive enough to declare (referential integrity for graph-like references, numeric ranges, and any other application level invariants that are desired)
 
-Note that while many apps might do mostly one or the other, a reasonable design pattern is take one approach for some data, and the other approach for the rest.
-For example the core parts of a document might take a schema on write approach, but allow extensible metadata that is handled with schema on read
-(which is a great place for different applications using the same document to store their data if they want to be less formal about versioning, and do best effort interop with each-other's metadata).
+Note that while many apps might do mostly one or the other, a reasonable design pattern is to take one approach for some data, and the other approach for the rest.
+For example, the core parts of a document might take a schema-on-write approach but allow extensible metadata that is handled with schema-on-read. This is a great place for different applications using the same document to store their data if they want to be less formal about versioning, and do best effort interop with each other's metadata.
 
 # Example/Proposed Design
 
 This design shows an easy to implement MVP, and is not intended to have document or API compatibility with what we do longer term.
-It minimally implements the requirements from "What options to support" while being extensible in the future.
-It can be subsetted for use in earlier milestones, and could be extended for later ones, see "Schedule" below.
+
+It minimally implements the requirements from [Options to Support](#options-to-support) while being extensible in the future.
+It can be subsetted for use in earlier milestones, and could be extended for later ones, see [Schedule](#schedule).
 
 This design is intentionally minimal while still being useful.
 Features not needed for minimal strongly typed application use or the JSON and XML domain schema are not included.
 
 ## Schema Representation
 
-The MVP should include a simple schema system, usable as a MVP for both `stored schema` and `view schema`.
+The MVP should include a simple schema system, usable as an MVP for both `stored schema` and `view schema`.
 
 One possible such schema system is included in [Schema.ts](./Schema.ts).
 
 These `TreeSchema` and `FieldSchema` can be added as `stored schema` as part of an edit op,
-which is considered conflicted if it tries to add a type that has a conflicting `name` and is not equal to the existing one.
+which is considered conflicted if it tries to add a type that has a conflicting `name` and is not equal to the existing one. TODO: This was confusing to me. This might be a bit more clear as part of the "Use as `Stored Schema` section.
```

Comment:
Question here

Reply:
@CraigMacomber any suggestions on this?

Reply:
Ya, this whole section is pretty horrible. Refactoring most of it into the more specific sections below would probably be better.